civihr_employee_portal/js/reports.js

```diff
@@ -926,6 +926,74 @@
             }.bind(this)
         });
 
+        $.ajax({
+            type: 'GET',
+            url: this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' ),
+            success: function(data) {
+                var mappedResults = normaliseSummaryData(data.results);
+
+                replaceSummaryTable(mappedResults, this.$DOM);
+            }.bind(this)
+        });
+
+        function normaliseSummaryData(summary) {
+            var normalisedSummary = {},
+                i;
+
+            // The data can have two formats, depending on particular report:
+            // data = { results: [{ departament: "xxx", count: 1 }, ...]) }
+            // or
+            // data = { results: [{ gender: "aaa" }, { gender: "aaa" }, ...] }
+            //
+            // The code inside if takes care of the first possibility
+            // The else takes care of the second
+            if (summary[0] !== undefined && summary[0].data.count !== undefined) {
+                for(i in summary) {
+                    normalisedSummary[summary[i].data.department] = +summary[i].data.count;
+                }
+            } else {
+                normalisedSummary = summary.map(function (row) {
+                    return { y: row.data.gender, x: 1 }
```

Comment:
The hardcoded properties I used are the actual property names that are used. They probably should have names like "xaxis" or "yaxis", but that would require changing all the report-related drupal views and fixing a lot of javascript, so I'm leaving it as is, at least for now.